src/core/pointcloud/qgslazinfo.cpp

```diff
@@ -0,0 +1,300 @@
+/***************************************************************************
+                         qgslazinfo.cpp
+                         --------------------
+    begin                : April 2022
+    copyright            : (C) 2022 by Belgacem Nedjima
+    email                : belgacem dot nedjima at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgslazinfo.h"
+
+#include "qgslogger.h"
+#include "qgsblockingnetworkrequest.h"
+#include "lazperf/readers.hpp"
+
+// QgsLazInfo
+
+QgsLazInfo::QgsLazInfo( std::ifstream &file )
+{
+  if ( !file.is_open() || !file.good() )
+  {
+    mIsValid = false;
+    return;
+  }
+  // TODO: add exceptions handling
+  long long previousPos = file.peek();
+  lazperf::reader::generic_file f( file );
+  mHeader = f.header();
+
+  parseHeader( mHeader );
+
+  file.seekg( 375 );
+  std::unique_ptr<char> vlrDataRaw( new char[ mPointRecordsOffset - 375 ] );
+  file.read( vlrDataRaw.get(), mPointRecordsOffset - 375 );
+
+  parseVlrData( vlrDataRaw.get(), mPointRecordsOffset - 375 );
+
+  parseCrs();
+
+  parseAttributes();
+
+  file.seekg( previousPos );
+  mIsValid = true;
+}
+
+QgsLazInfo::QgsLazInfo( QUrl &url )
```

Comment:
As discussed, I would prefer not to have this class do any network requests, only process byte arrays that are provided to it. I would suggest that:
- there is just a single constructor that initialized empty class
- there is a call to parse header from byte array
- there is another call to parse VLR data & everything else

We can have some utility functions to get those byte arrays from std::ifstream or from QgsBlockingNetworkRequest.